python_modules/libraries/dagster-gcp/dagster_gcp_tests/gcs_tests/test_io_manager.py

```diff
@@ -142,3 +148,26 @@ def dynamic():
         run_config={"resources": {"io_manager": {"config": {"gcs_bucket": gcs_bucket}}}}
     )
     assert result.success
+
+
+def test_asset_io_manager(gcs_bucket):
+    @asset
+    def upstream():
+        return 2
+
+    @asset
+    def downstream(upstream):
+        return 1 + upstream
+
+    asset_group = AssetGroup(
+        [upstream, downstream],
+        resource_defs={"io_manager": gcs_pickle_asset_io_manager, "gcs": gcs_resource},
```

Comment:
I'm considering making a mock gcs resource for unit testing. it's a bit odd to me to have a unit test interact with an external service, is this something you have discussed before?

Reply:
@alangenfeld - do you recall the history of this? Having a mock gcs resource sounds reasonable to me.

Reply:
Our gcp library gets less love in general than AWS, so wouldn't be surprised if it was just that we never built out that infra for gcp. We have those mocks for AWS.

Reply:
@sryza @dpeng817 added a first pass at a mock gcs client that can be wrapped into a resource. I only implemented the methods necessary for using the gcs io manager, but can take a look at adding more during a second pass.

Reply:
>  do you recall the history of this?

it may have been added in haste, but there is no good reason for it

> is this something you have discussed before?

"we should fix that" has been said a few times

> added a first pass at a mock gcs client

awesome, thank you